beanie/odm/documents.py

```diff
@@ -105,19 +103,23 @@
 if IS_PYDANTIC_V2:
     from pydantic import model_validator
 
-if TYPE_CHECKING:
-    from pydantic.typing import AbstractSetIntStr, DictStrAny, MappingIntStrAny
 
 DocType = TypeVar("DocType", bound="Document")
 DocumentProjectionType = TypeVar("DocumentProjectionType", bound=BaseModel)
 
 
 def json_schema_extra(schema: Dict[str, Any], model: Type["Document"]) -> None:
-    props = {}
-    for k, v in schema.get("properties", {}).items():
-        if not v.get("hidden", False):
-            props[k] = v
-    schema["properties"] = props
+    # remove excluded fields from the json schema
+    properties = schema.get("properties")
```

Comment:
I don't remember how this call is implemented in Pydantic. Could you please check if the schema was not given as a mutable object here? If so, modifying it can lead to unexpected errors - it is better to clone it then.
I'll double-check this point from my side too.

Reply:
Not exactly sure what you mean, `schema` is a mutable object (dict) as shown in the annotation. The `json_schema_extra` callable returns None so it's only called for its side-effects, mutating schema. It would have no effect if the schema was cloned. 

The calling code in Pydantic v2 is here: https://github.com/pydantic/pydantic/blob/main/pydantic/json_schema.py#L1364-L1368

Reply:
Yep, you are right. This is expected behavior.